### ISSUE TITLE
Tracking issue for `super_or_self_in_global_path` compatibility lint

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#36888]
#### What is this lint about

Paths starting with `self` or `super` are relative by definition, "global" prefix `::` used to indicate absolute paths doesn't make sense on them.
However, older versions of the compiler sometimes (e.g. in imports) accepted and ignored such prefixes:

```
mod m {
    struct S;

    use ::self::S as Z; // <- was accepted, the path was still interpreted as relative despite `::`
}
```

https://github.com/rust-lang/rust/pull/32403 fixed this oversight.
#### How to fix this warning/error

Remove `::` from the path starting with `::self` or `::super`.
#### Current status
- [x] https://github.com/rust-lang/rust/pull/32403 introduces the `super_or_self_in_global_path` lint as warn-by-default
- [x] https://github.com/rust-lang/rust/pull/36894 makes the `super_or_self_in_global_path` lint deny-by-default
- [ ] PR ? makes the `super_or_self_in_global_path` lint a hard error
